File: 547 Number of Provinces/Number of Provinces.py
#Python3 versiyonunu seçmeyi unutma =)
class Solution:
    def findCircleNum(self, array: List[List[int]]) -> int:
        n = len(array)
        counter =0
        checker = [0] * n

        def dfs(node):
            checker[node] = True

            for i in range(0,n):
                if array[node][i] == 1 and not checker[i]:
                    dfs(i)


        for index in range(0 , n):
            if not checker[index]:
                counter +=1
                dfs(index)
        
        return counter

File: 547 Number of Provinces/Number of Provinces.java
class Solution {
    public int findCircleNum(int[][] array) {
        int n = array.length;
        int counter = 0;
        int[] checker = new int[n];

        for (int index = 0; index < n; index++) {
            if (checker[index] == 0) {
                counter++;
                dfs(index, array, checker);
            }
        }

        return counter;
    }

    private void dfs(int node, int[][] array, int[] checker) {
        checker[node] = 1;

        for (int i = 0; i < array[node].length; i++) {
            if (array[node][i] == 1 && checker[i] == 0) {
                dfs(i, array, checker);
            }
        }
    }
}
